Sources/SwiftConcurrentSequence/Sequence+ConcurrentReduce.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  Created by Dan Federman on 4/30/22.
+//  Copyright © 2022 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS"BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Dispatch
+
+extension Sequence {
+
+    /// Returns the result of combining the elements of the sequence using the
+    /// given closure. The given closure is executed concurrently
+    /// on multiple queues to reduce the wall-time consumed by the reduction.
+    ///
+    /// Use the `reduce(into:_:)` method to produce a single value from the
+    /// elements of an entire sequence. For example, you can use this method on an
+    /// array of integers to filter adjacent equal entries or count frequencies.
+    ///
+    /// - Parameters:
+    ///   - defaultValue: A default value for Element. This value is utilized only
+    ///     when the receiver is empty.
+    ///   - reducingIntoFirst: A closure that combines an accumulating value and
+    ///     an element of the sequence into a new reduced value, to be used
+    ///     in the next call of the `reducingIntoFirst` closure or returned to
+    ///     the caller.
+    /// - Returns: The final reduced value. If the sequence has no elements,
+    ///   the result is `defaultValue`.
+    public func concurrentReduce(
+        defaultValue: @autoclosure () -> Element,
+        reducingIntoFirst: (inout Element, Element) -> ())
+    -> Element
```

Comment:
It's worth comparing this method signature to Apple's [reduce(into:_:)](https://developer.apple.com/documentation/swift/sequence/3128812-reduce).

Note that:
1. Apple's method reduces into a separate value, whereas this method reduces only the elements in the receiver.
2. Apple's method can change the returned type, whereas we are utilizing `Element` everywhere.

Both of these differences were created in order to aid parallelization. The recommended approach to concurrently achieving both a reduction and a type change is to execute a `concurrentMap` to change the sequence type and then execute `concurrentReduce` to reduce the sequence into a single instance.